src/sample_pool.rs

```diff
@@ -905,7 +905,10 @@ fn main() {
 
 								if leading_zeros >= client_target + WEAK_BLOCK_RATIO_0S {
 									if client.submitted_header_hashes.try_insert(&sketch.header_prevblock, block_hash) {
-										weak_block_submitted(&*submitter_state, client_id, &sketch.user_tag_1, our_payout, &header, &new_txn, &sketch.extra_block_data, leading_zeros, client_target);
+										let block_target = utils::nbits_to_target(sketch.header_nbits);
```

Comment:
Why not run this in kafka_submitter's weak_block_submitted to avoid the extra arguments and calculations when we don't need them.